common/web/gesture-recognizer/src/trackedPath.ts

```diff
@@ -29,10 +29,22 @@ namespace com.keyman.osk {
    *
    * `'complete'`: the touchpoint is no longer active; a touch-end has been observed.
    *   - Provides no parameters.
+   *   - Will be the last event raised by its instance, after any final 'segmentation'
+   *     events.
    *
    * `'invalidated'`: the touchpoint is no longer active; the path has crossed
    * gesture-recognition boundaries and is no longer considered valid.
    *   - Provides no parameters.
+   *   - Will precede recognition Promise fulfillment on the `Segment` provided by
+   *     the most recently-preceding 'segmentation' event.
```

Comment:
And shortly after this change, I realized that order doesn't matter, as `Promise` fulfillment and rejection are always async.  No need to make a distinction.